OSDOCS-4903: v2 PM pre-release comments nits (#55419)
openshift/openshift-docs: microshift_getting_started/microshift-architecture.adoc

[id="microshift-architecture"]
= Architecture
include::_attributes/attributes-microshift.adoc[]
include::_attributes/common-attributes.adoc[]
:context: microshift-architecture

toc::[]

Learn the specifics of {product-title} architecture including design intent, how it differs from {oke}, and API compatibility.

[id="microshift-architectural-design_{context}"]
== Architectural design

> {product-title} is a single-node container orchestration runtime designed to extend the benefits of using containers for running applications to low-resource edge environments. Because {product-title} is primarily as a platform for deploying applications, only the APIs and features essential to operating in edge and small form factor computing environments are included.

Reviewer: ```suggestion
{product-title} is a single-node container orchestration runtime designed to extend the benefits of using containers for running applications to low-resource edge environments. Because {product-title} is primarily a platform for deploying applications, only the APIs and features essential to operating in edge and small form factor computing environments are included.
```

> For example, {product-title} is tasked with only the following Kubernetes cluster capabilities:

Reviewer: suggestion to remove past tense:
```suggestion
For example, {product-title} contains only the following Kubernetes cluster capabilities:
```

* Networking
* Ingress
* Storage
* Helm

{product-title} also provides the following Kubernetes functions:

* Orchestration
* Security

> {product-title} is optimized when used with a compatible operating system, such as {op-system-ostree-first}. When used together, {product-title} and {op-system-ostree-first} form {op-system-bundle}. Virtual machines are handled by the operating system in {product-title} deployments.

Reviewer: More opportunities in this paragraph for reducing past tense

> .{product-title} as part of {op-system-bundle}.

Reviewer: The second attribute here, `{op-system-bundle}`, is not firing right in the HTML preview

> image::311_RHDevice_Edge_Overview_0223_1.png[<{product-title} is tasked with only the Kubernetes cluster services networking, ingress, storage, helm, with additional Kubernetes functions of orchestration and security.>]

Reviewer: I think this alt text could be improved to be more descriptive of the image. Something like "A multi-box diagram of Red Hat Device Edge with different boxes corresponding to different parts of the architecture. etc etc". This diagram is pretty complex, and i'm not sure what the user is supposed to focus on here. If it's the MicroShift component, the alt text could call out that box specifically and describe what it contains

Here are some links that were given to me about this, passing them on! 😄 

- https://www.w3.org/WAI/tutorials/images/complex/
- https://sc.edu/about/offices_and_divisions/digital-accessibility/guides_tutorials/alternative_text/charts-diagrams/index.php ("Example of Describing a Diagram")

Reviewer: also, what are the `<` and `>` for?

Author: I think the `<` and `>` place the caption? Just picked it up from the repo.

Author: The alt text is part of the file, not the caption. It should be there, but I can't see it now that I'm using the ones from graphics.

The following operational differences from {oke} can help you understand where {product-title} can be deployed:

[id="microshift-differences-oke_{context}"]
== Key differences from {oke}

* Devices with {product-title} installed are self-managing
* Compatible with RPM-OStree-based systems
* Uses only the APIs needed for essential functions, such as security and runtime controls
* Enables a subset of commands from the OpenShift CLI (`oc`) tool

> * Does not support workload high availability (HA) or horizontal scalability by the adding of worker nodes

Reviewer: minor suggestion:
```suggestion
* Does not support workload high availability (HA) or horizontal scalability with the addition of worker nodes
```

.{product-title} differences from {oke}.

> image::311_RHDevice_Edge_Overview_0223_2.png[<{product-title} is tasked with only the Kubernetes cluster capabilities of networking, ingress, storage, helm, with the additional Kubernetes functions of orchestration and security.>]

Reviewer: Same comment as preview image

Figure 2 shows that {oke} has the same cluster capabilities as {product-title}, and adds the following:

* Install
* Over-the-air updates
* Cluster Operators
* Operator Lifecycle Manager
* Monitoring
* Logging
* Registry
* Authorization
* Console
* Cloud Integration

> * Virtual Machines (VMs) through OpenShift Virtualization

Reviewer: ```suggestion
* Virtual Machines (VMs) through {VirtProductName}
```

The following operational differences from {OCP} can help you understand where {product-title} can be deployed:

[id="microshift-differences-ocp_{context}"]
== Key differences from {OCP}

* Devices with {product-title} installed are self-managing
* Compatible with RPM-OStree-based systems
* Uses only the APIs needed for essential functions, such as security and runtime controls
* Enables a subset of commands from the OpenShift CLI (`oc`) tool
* Does not support workload high availability (HA) or horizontal scalability by the adding of worker nodes

.{product-title} differences from {oke}.
image::microshift-RHDE-v-OKE.png[<{product-title} is tasked with only the Kubernetes cluster capabilities of networking, ingress, storage, helm, with the additional Kubernetes functions of orchestration and security.>]

Figure 2 shows that {oke} has the same cluster capabilities as {product-title}, and adds the following:

* Install
* Over-the-air updates
* Cluster Operators
* Operator Lifecycle Manager
* Monitoring
* Logging
* Registry
* Authorization
* Console
* Cloud Integration

> * Virtual Machines (VMs)

Reviewer 2: Virtual Machines (VMS) are also on the left hand side with MicroShift, but using a different technology. MicroShift uses KVM from RHEL (note that on left hand side lower blue box, there is a "VMs" text). Whereas OpenShift uses "OpenShift Virtualization" to run Virtual Machines. So hey have the same capbility to run VMs, but the management, configuration etc. of these is different (that is also denoted by the VMs ..... line on the left hand side).

I have no good idea how to put this complicated situation into nice text, sorry.

Author: Let's try breaking the thought up. Try lines 28 and 59ish in the update.

In {oke} and other {OCP} deployments, all of the components from the operating system through the cluster capabilities work as one comprehensive unit, with full cluster services for a multi-node Kubernetes workload. With {product-title}, functions such as over-the-air-updates, monitoring, and logging, are performed by the operating system.

> [id="microshift-openshift-apis_{context}"]

Reviewer 2: I still think that his section on APIs does not belong to a "Getting stared Architecture" section. I would suggest to remove it here and move it for example to the CLI tools and API reference sections.

Author: I agree. I will move it when I set up that section--not a simple operation. The CLI tools and API reference sections are generated straight from the OCP docs, so I have to make a new book for this content in our docs. With the current set-up, I am not able to move this content there. I want customers to have the information, so this is sort of parked here for the moment. These are high-priority sections and are next on my list.
